src/components/TracePage/TraceTimelineViewer/index.js

```diff
@@ -35,63 +37,36 @@ export default class TraceTimelineViewer extends Component {
     super(props);
 
     const { textFilter, xformedTrace } = props;
-    const filteredSpans = textFilter ? getFilteredSpans(xformedTrace, textFilter) : null;
-    this.state = {
-      filteredSpans,
-      collapsedSpans: new Set(),
-      endX: 100,
-      selectedSpans: new Set(),
-      startX: 0,
-      trace: props.xformedTrace,
-    };
-    this.toggleSpanCollapse = this.toggleSpanCollapse.bind(this);
-    this.toggleSpanSelect = this.toggleSpanSelect.bind(this);
+
+    this.store = createStore(reducer, newInitialState(xformedTrace));
```

Comment:
Still a little iffy on this. I would just reuse the store and create a new namespace

Reply:
Thanks. I've folded the state management into the top-level redux store.